pkg/autoscaler/provider.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2019 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package autoscaler
+
+import (
+	"errors"
+	"math"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/kubernetes-incubator/custom-metrics-apiserver/pkg/provider"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/metrics/pkg/apis/custom_metrics"
+)
+
+var concurrencyMetricInfo = provider.CustomMetricInfo{
+	GroupResource: v1alpha1.SchemeGroupVersion.WithResource("revision").GroupResource(),
```

Comment:
I believe this can just be `v1alpha1.Resource("revisions")` (note: "resource" is typically lower _and plural_)

Reply:
I wonder if we should make this `podautoscalers` though?  Feels like a bit of a layering violation to make this `revisions`.

Reply:
I even thought about making this „Pods“ but I‘m not yet 100% clear on the implications (it changes the API surface to be used on the HPA)